src/Spice86.Core/Emulator/InterruptHandlers/Input/Keyboard/BiosKeyboardBuffer.cs

```diff
@@ -68,7 +68,7 @@ public class BiosKeyboardBuffer : MemoryBasedDataStructure {
             return false;
         }
 
-        UInt16[tail] = code;
+        _indexable.UInt16[0, tail] = code;
```

Comment:
```suggestion
        _memory.UInt16[TailAddress] = code;
```

Reply:
Done, also renamed method AdvancePointer to ComputeNextAddress since it was not modifying anything